pyccel/parser/semantic.py

```diff
@@ -2139,7 +2139,8 @@ def _visit_FunctionalFor(self, expr, **settings):
         indices = expr.indices
         dims    = []
         body    = expr.loops[1]
-
+        #TODO remove the line below after adding equivalent of atoms to Basic issue #694
+        indices = [sp_Symbol(i.name) for i in indices]
```

Comment:
This is used in the big for loop later. `atoms` is one of the reasons that sympy is needed here, but it is not the only one. There is also `simplify` and `Summation`. As these two operations are purely sympy calculations it makes sense to keep the operations in this loop as sympy operations. That way there are fewer conversions between pyccel and sympy. So I think this TODO can be removed